### ISSUE TITLE
Необходима возможность использования КВАСа для клиента подключенного по SSTP к роутеру.

### ISSUE THREAD

**Issue by Paveltsibirna** [qzeleza/kvas#109]
**Опишите проблему.**
Необходима возможность использования КВАСа для клиента подключенного по SSTP к роутеру.

**Какие шаги, привели к ошибке/проблеме?**

1. На роутере настроил VPN-сервер SSTP для клиента.
2. Подключаюсь к роутеру по SSTP и телефона через приложение Open SSTP Client
3. Открываются все сайты которые не в списке КВАС, ip адрес интернета используемого на роутере, то есть маршрутизация есть, кроме маршрутизации через КВАС

**Какие ожидания о поведении программы?**
При подключении к роутеру клиента по SSTP открытие сайтов находящихся в списке КВАС (kvas list)

**Заполните информацию о системе на роутере:**
- Браузер chrome 121.0.6167.161
- SSH Putty 0.73

**Заполните информацию о самом роутере (пожалуйста, заполните следующую информацию):**
- Keenetic Ultra
-  версия 4.07 (на прошлых версиях такая же ситуация была)
-  KN-1811

**Comment by qzeleza** [comment]
Здесь не простая ситуация, так как sstp это не просто соединение в терминах rci, а это отдельная серверная надстройка, потому подход там должен быть почти такой же как и при ikev2.
Пока не готов говорить о сроках выпуска Кваса под данный тип соединения. 
Скорее всего может появиться в версии 1.2

**Comment by Paveltsibirna** [comment]
Если нужен будет доступ, дам...
Еще вопрос не совсем про это про wireguard потому как его начали блокировать,
можно ли где то в конфигурационных файлах сделать:
https://sysattack.com/posts/bypass-rkn-vpn/
https://gist.github.com/httpsx/76a98ea28e6f3a4ffc947e768c0b6c01

**Comment by qzeleza** [comment]
Прошу Вас подобного рода ссылки не публиковать (удалил намерено). 

Может быть в версии 1.2.

**Comment by Paveltsibirna** [comment]
тоже думал не публиковать. хорошо что удалили, но как передать всю боль :)

**Comment by Paveltsibirna** [comment]
Еще вопрос, есть ли возможность использования не только adguard, а например по выбору с uBlockOrigin

**Comment by qzeleza** [comment]
> Еще вопрос, есть ли возможность использования не только adguard, а например по выбору с uBlockOrigin

Поясните пожалуйста

**Comment by qzeleza** [comment]
> в диагностике проскакивают вот такие штуки Фев 9 19:03:39 КВАС ОШИБКА::\033[1;31m[ip4_firewall_dns_rules_set] Возникла ошибка в функции при установке правил iptables.\033[m Фев 9 19:03:39 ndm Opkg::Manager: /opt/etc/ndm/netfilter.d/100-vpn-mark: sh: bad number.

О какой версии Кваса идет речь?

**Comment by Paveltsibirna** [comment]
Последняя, могу сюда дебаг закинуть

**Comment by qzeleza** [comment]
Нет, тут дебаг не нужен - ошибка явная и описана в отладочном сообщении. 
Посмотрю в чем дело. Спасибо.

**Comment by Paveltsibirna** [comment]
> > Еще вопрос, есть ли возможность использования не только adguard, а например по выбору с uBlockOrigin
> 
> Поясните пожалуйста

ну есть uBlockOrigin, она от рекламы избавляется лучше. Хотелось бы что бы этот инструмент был в КВАСЕ, потом когда нибудь :)

**Comment by qzeleza** [comment]
> ну есть uBlockOrigin, она от рекламы избавляется лучше. Хотелось бы что бы этот инструмент был в КВАСЕ, потом когда нибудь :)

Для этой цели есть **adblock** в самом Квасе.

**Comment by Drafteed** [comment]
> **sstp** это не просто соединение в терминах **rci**

Подскажите, с Wireguard сервером на роутере будет работать маршрутизация через Квас и далее через VPN клиент на роутере? Сейчас использую решение https://github.com/DennoN-RUS/Bird4Static хочу перейти на Ваше. И пользуясь случаем хотел бы спросить насчет поддержки BGP, планируется ли? Это очень удобно для заворачивания больших сетей, типа меты.

**Comment by qzeleza** [comment]
Вы имеете ввиду  смогу ли клиенты Wireguard сервера использовать Квас?
Честно говоря не нижу в том проблем. Но не проверял - на все 100% не уверен.

По поводу BGP - точно нет.

**Comment by nobbynobbs** [comment]
@Paveltsibirna 

Трям, здравствуйте!

```sh
iptables -A PREROUTING -w -t nat -i sstp+ -p tcp -m set --match-set unblock dst -j REDIRECT --to-ports 1181
iptables -A PREROUTING -w -t nat -i sstp+ -p udp -m set --match-set unblock dst -j REDIRECT --to-ports 1181
```

Это две команды которые нужно выполнить в терминале, они создадут два правила которые будут делать переадресацию пакетов из sstp куда надо.

За почти готовое решение спасибо форуму кинетика и вот этому [сообщению](https://forum.keenetic.com/topic/14415-%D0%BF%D1%80%D0%BE%D0%B1%D1%83%D0%B5%D0%BC-%D0%BA%D0%B2%D0%B0%D1%81/?do=findComment&comment=165884).

Возможно я чего-то не понимаю (а я в маршрутизации почти ничего не понимаю) и переупрощаю, но у себя настроил и вроде бы оно работает как надо.

**Comment by Paveltsibirna** [comment]
Спасибо! проверю, может еще уважаемый [qzeleza](https://github.com/qzeleza) обратит на это внимание.

**Comment by qzeleza** [comment]
> Это две команды которые нужно выполнить в терминале

Доброго дня
Возможно эти две команды и решают вопросы, но только для **ssr**

Относительно остальных **vpn** в режимах **fastnet** и без оного  - есть идеи?

**Comment by nobbynobbs** [comment]
@qzeleza здравствуйте, спасибо за проект.

> Возможно эти две команды и решают вопросы, но только для ssr

Да, извиняюсь что не уточнил в первом сообщении, этот ad-hoc действительно только для маршрутизации sstp -> ssr, а не универсальное решение. @Paveltsibirna FYI

> Относительно остальных vpn в режимах fastnet и без оного - есть идеи?

Увы, нет.

**Comment by Drafteed** [comment]
> Вы имеете ввиду смогу ли клиенты Wireguard сервера использовать Квас?
> Честно говоря не нижу в том проблем. Но не проверял - на все 100% не уверен.

Подтверждаю, работает. В конфигурации клиента WG в качестве DNS нужно указать адрес IP адрес роутера внутри туннеля (например у меня `192.168.2.1`). Кроме этого в `/opt/etc/dnsmasq.conf`, нужно добавить новый `listen-address` с этим IP адресом. Либо вообще убрать все строчки `listen-address`, чтобы `dnsmasq` слушал все интерфейсы.

**Comment by ALTernateF13** [comment]
> Относительно остальных **vpn** в режимах **fastnet** и без оного - есть идеи?

Т.к. это отдельный сетевой интерфейс `sstp+`, то всё должно работать по аналогии с гостевым WiFi `br1`.

> Подтверждаю, работает

А не затруднит приложить вывод ifconfig? Или просто подскажите, какое имя у сетевого интерфейса для Wireguard — `wg0`?

**Comment by Paveltsibirna** [comment]
То есть должно работать ?
что нужно сделать ?

**Comment by AltGrF13** [comment]
> Здесь не простая ситуация, так как **sstp** это не просто соединение в терминах **rci**, а это отдельная серверная надстройка, потому подход там должен быть почти такой же как и при **ikev2**. Пока не готов говорить о сроках выпуска **Кваса** под данный тип соединения. Скорее всего может появиться в версии **1.2**

Из праздного любопытства был дёрнут API-метод `curl -s 127.0.0.1:79/rci/sstp-server`, дык у них даже какого-то единого стандарта ответа с IKEv2 нет, всё очень не консистентно:
```
  "interface": "Bridge0",
  "pool-range": {
    "begin": "192.168.4.35",
    "size": "115"
  },
  "multi-login": true,
  "lcp": {
    "echo": {
      "interval": 30,
      "count": 3
    }
  }
```

```
  "enable": true,
  "nat": true,
  "pool-start": "192.168.2.35",
  "pool-size": "115",
  "dns-server": "192.168.1.1",
  "multi-login": true,
  "sa-compat": "default"
```

**Comment by Ajivitca** [comment]
> Здесь не простая ситуация, так как **sstp** это не просто соединение в терминах **rci**, а это отдельная серверная надстройка, потому подход там должен быть почти такой же как и при **ikev2**. Пока не готов говорить о сроках выпуска **Кваса** под данный тип соединения. Скорее всего может появиться в версии **1.2**

так же проблема удаленной настройки на серых адресах. тоже только sstp работает без танцев с бубном.

**Comment by AltGrF13** [comment]
В [бете](https://t.me/kvas_pro/3310/20276) уже больше месяца  можно включать SSTP через `kvas route`.
Спустя сколько-то сборок появится и удобная команда добавления именно SSTP-сети, а не просто её IP диапазона. Поэтому вопрос можно закрывать.

**Comment by Paveltsibirna** [comment]
Все работает! Спасибо!
